packages/blade/src/components/Input/BaseInput/baseInputStyles.ts

```diff
@@ -0,0 +1,111 @@
+import { getPlatformType, makeBorderSize, makeSpace } from '~utils';
+
+type GetInputStyles = Pick<
+  BaseInputProps,
+  | 'isDisabled'
+  | 'validationState'
+  | 'leadingIcon'
+  | 'prefix'
+  | 'interactionElement'
+  | 'suffix'
+  | 'trailingIcon'
+> & {
+  isFocused?: boolean;
+  theme: Theme;
+};
+
+export const getInputBackgroundAndBorderStyles = ({
+  theme,
+  isFocused,
+  isDisabled,
+  validationState,
+}: GetInputStyles): CSSObject => {
+  // normal state
+  let backgroundColor = theme.colors.brand.gray[200];
+  let borderBottomColor = theme.colors.brand.gray[400];
+
+  // focused state
+  if (isFocused) {
+    backgroundColor = theme.colors.brand.primary[300];
+    borderBottomColor = theme.colors.brand.primary[500];
+  }
+
+  // disabled state
+  if (isDisabled) {
+    backgroundColor = theme.colors.brand.gray[200];
+    borderBottomColor = theme.colors.brand.gray[300];
+  }
+
+  // validation state
+  if (validationState === 'error') {
+    backgroundColor = theme.colors.feedback.background.negative.lowContrast;
+    borderBottomColor = theme.colors.feedback.border.negative.highContrast;
+  } else if (validationState === 'success') {
+    backgroundColor = theme.colors.feedback.background.positive.lowContrast;
+    borderBottomColor = theme.colors.feedback.border.positive.highContrast;
+  }
+
+  return {
+    backgroundColor,
+    borderBottomColor,
+    borderTopLeftRadius: makeBorderSize(theme.border.radius.small),
+    borderTopRightRadius: makeBorderSize(theme.border.radius.small),
+    borderBottomWidth: makeBorderSize(theme.border.width.thin),
+    borderBottomStyle: 'solid',
+  };
+};
+
+export const getBaseInputStyles = ({
+  theme,
+  isDisabled,
+  leadingIcon,
+  prefix,
+  interactionElement,
+  suffix,
+  trailingIcon,
+}: GetInputStyles): CSSObject => {
+  const isReactNative = getPlatformType() === 'react-native';
+
+  const {
+    hasLeadingIcon,
+    hasPrefix,
+    hasInteractionElement,
+    hasSuffix,
+    hasTrailingIcon,
+  } = getInputVisualsToBeRendered({
+    leadingIcon,
+    prefix,
+    interactionElement,
+    suffix,
+    trailingIcon,
+  });
+
+  return {
+    ...getTextStyles({
+      size: 'medium',
+      variant: 'body',
+      type: isDisabled ? 'placeholder' : 'subtle',
+      weight: 'regular',
+      contrast: 'low',
+      theme,
+    }),
+    // take the full available width of parent container for input field
+    flex: 1,
+    backgroundColor: 'transparent',
+    paddingTop: makeSpace(theme.spacing[2]),
+    paddingBottom: makeSpace(theme.spacing[2]),
+    paddingLeft:
+      hasLeadingIcon || hasPrefix ? makeSpace(theme.spacing[2]) : makeSpace(theme.spacing[3]),
+    paddingRight:
+      hasInteractionElement || hasSuffix || hasTrailingIcon
+        ? makeSpace(theme.spacing[2])
+        : makeSpace(theme.spacing[3]),
+    ...(isReactNative ? { lineHeight: undefined, height: '36px' } : {}),
+    ...(!isReactNative ? { outline: 'none', border: 'none' } : {}),
```

Comment:
🤔 

```ts
    ...(isReactNative
      ? { lineHeight: undefined, height: '36px' }
      : { outline: 'none', border: 'none' }),
```

Reply:
literally wrote this and then erased it 😄 

The thing was it was too confusing because the properties are diff hence thought to keep it 2 diff statements and anyways since I'm not re-computing i think readability > shorthand 🙊